devicetypes/widomsrl/widom-smart-dry-contact.src/widom-smart-dry-contact.groovy

```diff
@@ -0,0 +1,340 @@
+	}
+    
+	if (cmds) {
+		runIn(10, "syncCheck")
+		sendHubCommand(cmds,1000)
+	} else {
+		runIn(1, "syncCheck")
+	}
+}
+
+private syncCheck() {
+	logging("Executing syncCheck()","info")
+	def failed = []
+	def incorrect = []
+	def notSynced = []
+	parameterMap().each {
+		if (state."$it.key"?.state == "incorrect") {
+			incorrect << it
+		} else if (state."$it.key"?.state == "failed") {
+			failed << it
+		} else if (state."$it.key"?.state in ["inProgress","notSynced"]) {
+			notSynced << it
+		}
+	}
+
+	if (failed) {
+		multiStatusEvent("Sync failed! Verify parameter: ${failed[0].num}", true, true)
+	} else if (incorrect) {
+		multiStatusEvent("Sync mismatch! Verify parameter: ${incorrect[0].num}", true, true)
+	} else if (notSynced) {
+		multiStatusEvent("Sync incomplete! Open settings and tap Done to try again.", true, true)
+	} else {
+		if (device.currentValue("multiStatus")?.contains("Sync")) { multiStatusEvent("Sync OK.", true, true) }
+	}
+}
+
+private multiStatusEvent(String statusValue, boolean force = false, boolean display = false) {
+	if ( !device.currentValue("multiStatus")?.contains("Sync") || device.currentValue("multiStatus") == "Sync OK." || force ) {
+		sendEvent(name: "multiStatus", value: statusValue, descriptionText: statusValue, displayed: display)
+	}
+}
+
+private deviceIdEvent(String value, boolean force = false, boolean display = false) {
+	sendEvent(name: "deviceID", value: value, descriptionText: value, displayed: display)
+}
+
+//event handlers related to configuration and sync
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	def paramKey = parameterMap().find( {it.num == cmd.parameterNumber } ).key
+	logging("Parameter ${paramKey} value is ${cmd.scaledConfigurationValue} expected " + state."$paramKey".value, "info")
+	state."$paramKey".state = (state."$paramKey".value == cmd.scaledConfigurationValue) ? "synced" : "incorrect"
+	syncNext()
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejectedRequest cmd) {
+	logging("rejected request!","warn")
+	for ( param in parameterMap() ) {
+		if (state."$param.key"?.state == "inProgress") {
+			state."$param.key"?.state = "failed"
+			break
+		}
+	}
+}
+//event handlers
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	//ignore
+}
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	logging("SwitchBinaryReport received, value: ${cmd.value} ","info")
+	sendEvent([name: "switch", value: (cmd.value == 0 ) ? "off": "on"])
+}
+
+/*
+####################
+## Z-Wave Toolkit ##
+####################
+*/
+def parse(String description) {
+	def result = []
+	def deviceId = [];
+	logging("Parsing: ${description}")
+	if (description.startsWith("Err 106")) {
+	result = createEvent(
+				descriptionText: "Failed to complete the network security key exchange. If you are unable to receive data from it, you must remove it from your network and add it again.",
```

Comment:
Could You fix the indentation here ? (it seems that 3 tabs are enough here)

Reply:
fixed